### PR TITLE
[fix][builtin] polymorphic builtins shouldn't inherit attributes

### PR DESCRIPTION
**Pull request by zoecarver**
This PR makes all polymorphic builtins have empty attributes. Some attributes (such as read none) apply to the "regular" version of the builtin and not the polymorphic version. These builtins only exist in raw sil so, the error wasn't caught before. However, #30429 exposed the issue because mandatory combine runs during Onone. This is an NFC until #30429 lands.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

File: include/swift/AST/Builtins.def [hunk 1]
```diff
@@ -113,7 +113,7 @@ BUILTIN_CAST_OR_BITCAST_OPERATION(SExtOrBitCast,  "sextOrBitCast",  "n")
 #ifndef BUILTIN_BINARY_OPERATION_ALL
 #define BUILTIN_BINARY_OPERATION_ALL(Id, Name, Attrs, Overload) \
   BUILTIN_BINARY_OPERATION_OVERLOADED_STATIC(Id, BUILTIN_BINARY_OPERATION_GENERIC_HELPER_STR(Name), Attrs, Overload) \
-  BUILTIN_BINARY_OPERATION_POLYMORPHIC(Generic##Id, BUILTIN_BINARY_OPERATION_GENERIC_HELPER_STR(generic_##Name), Attrs)
+  BUILTIN_BINARY_OPERATION_POLYMORPHIC(Generic##Id, BUILTIN_BINARY_OPERATION_GENERIC_HELPER_STR(generic_##Name), "")
```
Comment (gottesmm) [inline include/swift/AST/Builtins.def]: I would rather change BUILTIN_BINARY_OPERATION_POLYMORPHIC to not take that flag at all.

Comment (zoecarver) [reply include/swift/AST/Builtins.def]: Good idea. I'll do that.

**Review by gottesmm — APPROVED** [review]
Please squash the commits. But beyond that, LGTM.

**Comment by zoecarver** [comment]
@swift-ci please smoke test and merge.

**Comment by zoecarver** [comment]
@swift-ci please smoke test and merge.

**Pull request merged**